include/roboteam_ai/world/Field.h

```diff
@@ -185,6 +213,34 @@ class Field {
     // The top most point of the opponents goal (this point is on the right line).
     std::optional<Vector2> theirTopGoalSide;
 
+    // The top position, point with the highest y-coordinate, of the left penalty line (our penalty line).
+    std::optional<Vector2> leftPenaltyLineTop;
```

Comment:
I don't really like that all these variables are stored separately while they are often unused. These are all copies whenever you get a new field message. Why not make your getters return references to the original proto message? This is all extra work and extra code.

Reply:
Well it is better to be consistent in your design/approach. All computations are done during construction of the Field class and all these computed values are stored. You cannot be inconsistent and store halve of the values whereas the other halve of the values is computed when the getter is executed.

Reply:
If you abstract it away under getters it is fine, you hide the (more efficient) solution anyway.